app/assets/javascripts/discourse/app/static/prosemirror/extensions/emoji.js

```diff
@@ -170,6 +170,7 @@ const extension = {
 
   serializeNode: {
     emoji(state, node) {
+      state.flushClose();
```

Comment:
I'm guessing no, but is there any way we can do this proactively for all of these custom nodes? Otherwise people will always need to remember to do this, which is just another thing to teach.

If not that's fine, just something to keep aware of

Reply:
Yeah this is mostly an inline node concern, especially the ones we can join with the surrounding characters without spaces. I don't think we should do this proactively for all nodes because of the extra work for blocks, but we definitely can abstract most of this work behind an easier helper.

I didn't explore this yet because this boundary check was initially an experiment, but it turned out to work well. An alternative for this approach would be doing it within the prosemirror document with a custom plugin, but it would be way more work with just one slight difference in behavior.